### PR TITLE
feat(iot): add WaitForHub IoT helper

### PR DESCRIPTION
**Pull request by didjcodt**
Signed-off-by: Clément Decoodt <cdecoodt@online.net>

This is highly inspired from k8s product WaitForCluster, adapted for IoT Hubs

### CODE REVIEW

File: api/iot/v1beta1/iot_helpers.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	HubID         string
+	Region        scw.Region
+	Status        HubStatus
+	Timeout       *time.Duration
+	RetryInterval *time.Duration
+}
+
+// WaitForHub waits for the hub to be in a ready state before returning.
+func (s *API) WaitForHub(req *WaitForHubRequest) (*Hub, error) {
+	timeout := *req.Timeout
```
Comment (remyleone) [inline api/iot/v1beta1/iot_helpers.go:27]: `*req.Timeout` might be nil

```suggestion
	timeout := defaultTimeout
	if req.Timeout != nil {
		timeout = *req.Timeout
	}
```

Comment (didjcodt) [reply api/iot/v1beta1/iot_helpers.go]: You're completely right, this is even done on the next parameter check. Fixed, thanks!

File: api/iot/v1beta1/iot_helpers.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		Get: func() (interface{}, bool, error) {
+			hub, err := s.GetHub(&GetHubRequest{
+				HubID:  req.HubID,
+				Region: req.Region,
+			})
+			if err != nil {
+				return nil, false, err
+			}
+
+			return hub, hub.Status == HubStatusReady, nil
```
Comment (jerome-quere) [inline api/iot/v1beta1/iot_helpers.go:46]: What is hub is in error or lock or disabled, these are final status too ?  Have a look at instance WaitServer on how we manage multiple terminal status

Comment (didjcodt) [reply api/iot/v1beta1/iot_helpers.go]: I updated to add these transient statuses too. For lock, this is not returned as a state but as an error in cases of writes on a locked hub.

**Pull request merged**